cmd/snap/cmd_run.go

```diff
@@ -33,10 +34,12 @@ import (
 
 type cmdRun struct {
 	Positional struct {
-		SnapApp string `positional-arg-name:"<app name>" description:"the application to run, e.g. hello-world.env"`
+		SnapApp string `positional-arg-name:"<app name>" description:"the snap (e.g. hello-world) or application to run (e.g. hello-world.env)"`
 	} `positional-args:"yes" required:"yes"`
 
-	Command string `long:"command" description:"alternative command to run"`
+	Command  string `long:"command" description:"alternative command to run"`
+	Hook     string `long:"hook" description:"run a specific hook contained within the snap"`
+	Revision string `long:"revision" description:"use a specific snap revision instead of the active one (this only applies when using --hook)"`
```

Comment:
Okay, we can use the hidden flag then, but let's do this:

- Hide it as you suggest.
- Call it `-r` instead of `--revision`. That's nicer if we decide to offer it as real user interface.
- Error out for the time being if -r is provided without --hook.